### PR TITLE
Support Card to Product script

### PR DESCRIPTION
**Pull request by ZeldaZach**
- Generates a mapping from card UUIDs to sealedProduct UUIDs that the specific card printing can be found in
- Will need to add GHA to update the repo with this, or something similar

[foo.json.txt](https://github.com/mtgjson/mtg-sealed-content/files/12270439/foo.json.txt)

### CODE REVIEW

**Comment by kodawah** [comment]
as a small improvement, we should consider `etched` property too, when the booster sheet name contains "etched" (because foil is a boolean right now)

**Comment by kodawah** [comment]
lgtm

it just need the etched fix here, where to check etched in sheet name, then foil property, then nonfoil

 >           finish = "foil" if sheet_data["sheets"][sheet]["foil"] else "nonfoil"

File: card_to_product_compiler.py [hunk 1]
```diff
@@ -170,7 +170,11 @@ def get_cards_in_pack(self, set_code: str, booster_code: str) -> List[Card]:
         return_value = set()
         for sheet in sheets_to_poll:
             cards_in_sheet = sheet_data["sheets"][sheet]["cards"]
-            finish = "foil" if sheet_data["sheets"][sheet]["foil"] else "nonfoil"
+
+            if "etched" in sheet:
```
Comment (kodawah) [inline card_to_product_compiler.py:174]: this check needs to be lowercase (ie `foilEtchedLegend`)

File: card_to_product_compiler.py [hunk 2]
```diff
@@ -0,0 +1,226 @@
...
+            else:
+                finish = "foil" if sheet_data["sheets"][sheet]["foil"] else "nonfoil"
+
+            for card_uuid in cards_in_sheet.keys():
+                return_value.add(Card(card_uuid, finish))
+
+        return list(return_value)
+
+    def get_cards_in_deck(self, set_code: str, deck_name: str) -> List[Card]:
+        decks_data = self.mtgjson_data[set_code].get("decks")
```
Comment (kodawah) [inline card_to_product_compiler.py:185]: we'd need to test `sealed` as well

File: card_to_product_compiler.py [hunk 3]
```diff
@@ -0,0 +1,226 @@
...
+                                "set": "w17"
+                            }
+                        ]
+                    }
+                ]
+            }
+            """
+            return_value = set()
+            for config in content["configs"]:
+                for deck in config.get("decks", []):
```
Comment (kodawah) [inline card_to_product_compiler.py:149]: we'd need to test every possible type but at least `sealed` as well

File: card_to_product_compiler.py (hunk 1 above)
Comment (ZeldaZach) [reply card_to_product_compiler.py:174]: Wilco

File: card_to_product_compiler.py (hunk 3 above)
Comment (ZeldaZach) [reply card_to_product_compiler.py:149]: I looked and only "decks" and "sealed" exist

Comment (axxroytovu) [reply card_to_product_compiler.py:149]: Just FYI I do have a variable/config/card already in the pipeline for the next update if you want to make sure that it compiles correctly.

Comment (ZeldaZach) [reply card_to_product_compiler.py:149]: I can preemptively add support for config "card", sure

Comment (kodawah) [reply card_to_product_compiler.py:153]: @axxroytovu is there a variable/config/booster planned too?

Comment (axxroytovu) [reply card_to_product_compiler.py:153]: It's possible based on the raw formatting but I don't know of any products that could realistically have that configuration.

File: outputs.py [hunk 4]
```diff
@@ -308,3 +313,7 @@ def deck_links(contents, uuid, logger=None):
 
 with open("outputs/products.json", "w") as outfile:
     json.dump(products_new, outfile)
+
+card_to_product_compiler.main(
+    argparse.Namespace(input_file="mtgJson/AllPrintings.json", output_file="outputs/card_to_products.json")
```
Comment (axxroytovu) [inline outputs.py:318]: Default output file should be `card_map.json` to match the existing convention.

**Review by axxroytovu — COMMENTED** [review]
Resolve conflicts with master before merge, otherwise good.

Edit: Never mind, looks like the issue is already resolved. Just waiting on other comments to be resolved.

**Pull request merged**